src/docs/asciidoc/index.adoc

```diff
@@ -815,6 +816,257 @@ and it provides tasks to generate a GraalVM native image and also creates the Gr
+logback.xml.to.java.enabled = true
+
+# Precomputes Micronaut configuration property keys from the current environment variables
+precompute.environment.properties.enabled = true
+
+# Scans reactive types at build time instead of runtime
+scan.reactive.types.enabled = true
+
+# Seals environment property values: environment properties will be deemed immutable after application startup.
+sealed.environment.enabled = true
+
+# Scans for service types ahead-of-time, avoiding classpath scanning at startup
+serviceloading.jit.enabled = true
+# The list of service types to be scanned (comma separated)
+service.types = io.micronaut.Service1,io.micronaut.Service2
+# A list of implementation types which shouldn't be included in the final application (comma separated)
+serviceloading.rejected.impls = com.Misc,org.Bar
+
+# Converts YAML configuration files to Java configuration
+yaml.to.java.config.enabled = true
+
+# Precomputes property sources at build time
+sealed.property.source.enabled = true
+----
+
+Another file, `build/generated/aot/samples/native/native.properties` will contain the same, but with the options which are relevant to an application compiled to a _native image_:
+
+----
+# Generates GraalVM configuration files required to load the AOT optimizations
+graalvm.config.enabled = true
+# The list of service types to be scanned (comma separated)
+service.types = io.micronaut.Service1,io.micronaut.Service2
+
+# Checks of existence of some types at build time instead of runtime
+known.missing.types.enabled = true
+# A list of types that the AOT analyzer needs to check for existence (comma separated)
+known.missing.types.list = javax.inject.Inject,io.micronaut.SomeType
+
+# Replaces logback.xml with a pure Java configuration (NOT YET IMPLEMENTED!)
+logback.xml.to.java.enabled = true
+
+# Precomputes Micronaut configuration property keys from the current environment variables
+precompute.environment.properties.enabled = true
+
+# Scans reactive types at build time instead of runtime
+scan.reactive.types.enabled = true
+
+# Seals environment property values: environment properties will be deemed immutable after application startup.
+sealed.environment.enabled = true
+
+# Scans for service types ahead-of-time, avoiding classpath scanning at startup
+serviceloading.native.enabled = true
+# The list of service types to be scanned (comma separated)
+service.types = io.micronaut.Service1,io.micronaut.Service2
+# A list of implementation types which shouldn't be included in the final application (comma separated)
+serviceloading.rejected.impls = com.Misc,org.Bar
+
+# Converts YAML configuration files to Java configuration
+yaml.to.java.config.enabled = true
+
+# Precomputes property sources at build time
+sealed.property.source.enabled = true
+----
+
+For native images, it is important to always have the `graalvm.config.enabled` option set to `true`, otherwise the AOT optimizations will not be loaded.
```

Comment:
How can we set that using the DSL?

Reply:
You don't, the plugin does it automatically.

Reply:
:+1: